coral-common/src/main/java/com/linkedin/coral/common/FuzzyUnionSqlRewriter.java

```diff
@@ -67,14 +68,16 @@
  *   SELECT *
  *   FROM a
  *   UNION ALL
- *   SELECT view_c.f, generic_project(view_c.f1, 'view_c.f1') as f1
+ *   SELECT view_c.f, generic_project(view_c.f1, 'view_c.f1', hive_type_string) as f1
```

Comment:
why do we need to change the semantics of generic_project? why cant the logic - `RelDataType expectedRelDataType = call.getType();
        String expectedRelDataTypeString = RelDataTypeToHiveTypeStringConverter.convertRelDataType(expectedRelDataType);`

be done in FuzzyUnionGenericProjectTransformer by utilizing `getRelDataType()` utility.

Reply:
I am trying to avoid using `getRelDataType()` in Coral-Spark, so that we may be able to achieve transformation migrations without changing client-facing API.
Besides, changing semantics of `generic_project` for coral-spark doesn't have side effects, actually the existing second parameter column name (like `'view_c.f1'` in this example) is for convenient transformation in Coral-Trino.